### PR TITLE
Flink SQL tutorial

### PR DESCRIPTION
**Pull request by davetroiano**

### CODE REVIEW

File: .semaphore/semaphore.yml
Comment (bbejeck) [inline .semaphore/semaphore.yml:470]: This needs to go in the `flinksql` section of the `.semaphore.yml` file

File: _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/expected-print-output-topic.log
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/expected-print-output-topic.log:2]: nit: are there any stock number formatters we could apply? These numbers are hard to look at - I realize that's not the point of the tutorial

File: _includes/tutorials/aggregating-minmax/flinksql/markup/dev/transient-query.adoc
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/markup/dev/transient-query.adoc:24]:
```suggestion
Or, as a third option, you can see streaming results non-materialized and inline in the SQL client by running ``SET 'sql-client.execution.result-mode' = 'tableau';`` and rerunning the query once more. In this case, the results will look very similar to `changelog` mode results. This is because tables sourced by the Kafka connector are _unbounded_ and can thus only be queried in `streaming` mode. For further reading on these Flink SQL concepts, consult the documentation on  https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/table/sqlclient/#sql-client-result-modes[SQL client result modes]  and https://nightlies.apache.org/flink/flink-docs-release-1.16/docs/dev/datastream/execution_mode/[streaming vs. batch execution]
```

**Review by bbejeck — COMMENTED** [review]
Excellent job, @davetroiano - I have a few minor comments, but overall, it LGTM.
NOTE: This was a text-only review. I still need to run the tutorial locally.

**Review by bbejeck — COMMENTED** [review]
I ran the tutorial and over all it LGTM modulo the comments.

File: docker/flink/Dockerfile.flink-kafka
Comment (davetroiano) [reply docker/flink/Dockerfile.flink-kafka:2]: right, not needed in the clone script. this is just the Dockerfile used to build the image in each tutorial's docker compose, but the building and publishing gets done out of band. it'll likely only get updated infrequently, e.g., for version upgrades.

File: docker/flink/Dockerfile.flink-sql-client-kafka
Comment (davetroiano) [reply docker/flink/Dockerfile.flink-sql-client-kafka:1]: same deal as above, not needed in the clone script

File: _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/expected-print-output-topic.log
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/expected-print-output-topic.log:2]: I agree... other SQL dialects have convenient built-in functions for this but I'm not (yet) finding anything in Flink SQL.  will keep looking

File: _includes/tutorials/aggregating-minmax/flinksql/markup/dev/create-movie-sales-by-year.adoc
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/markup/dev/create-movie-sales-by-year.adoc:1]: done

File: _includes/tutorials/aggregating-minmax/flinksql/markup/dev/validate-movie-sales-by-year.adoc
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/markup/dev/validate-movie-sales-by-year.adoc]: good call, done.

File: _includes/tutorials/aggregating-minmax/flinksql/markup/dev/validate-movie-sales-by-year.adoc
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/markup/dev/validate-movie-sales-by-year.adoc:14]: yes - done

File: _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/validate-movie-sales-by-year.sh
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/code/tutorial-steps/dev/validate-movie-sales-by-year.sh]: no, good call. removed this line

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java:151]: I remember the conversation about the need to do this. One thought would be to put the ref to the `ByteArrayOutputStream` in the thread and run a loop checking for results as a way to make the test a little more deterministic and eliminate flaky failures.  But this works as well.

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java:74]: It doesn't matter for this tutorial since there's only one test, but if we end up with more than one test we can put the container setup in a static block or a`@BeforeAll` so we do the container setup once for all tests - just a note for the future.

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java:38]:
```suggestion
  protected StreamTableEnvironment streamTableEnv;
```
Super nit - feel free to ignore

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlAggregatingMinMaxTest.java
Comment (bbejeck) [inline _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/FlinkSqlAggregatingMinMaxTest.java:20]: with schema registry when you set the port to something like `bogus` it will automatically use the `MockSchemaRegistry`, but I'm not sure how that would play out with the flink environment - so I'd leave this as is for now.

**Review by bbejeck — APPROVED** [review]
Nice work! I have a couple of optional comments - otherwise, it LGTM

**Comment by bbejeck** [comment]
FWIW the harness runner for the tutorial passed the the unit test failed
```
Task :test
io.confluent.developer.FlinkSqlAggregatingMinMaxTest > simpleSelect FAILED
    java.lang.reflect.InaccessibleObjectException at FlinkSqlAggregatingMinMaxTest.java:18
```

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java]: good idea, changed to do this rather than sleep

File: _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java
Comment (davetroiano) [reply _includes/tutorials/aggregating-minmax/flinksql/code/src/test/java/io/confluent/developer/AbstractFlinkKafkaTest.java]: agree, changed

**Pull request merged**
